client/consensus/epochs/src/lib.rs

```diff
@@ -515,6 +510,12 @@ where
 		number: Number,
 		slot: E::Slot,
 	) -> Result<(), fork_tree::Error<D::Error>> {
+		if self.gap.is_some() {
+			// Do not remove any node while a gap is present.
+			// Tree nodes may be required while importing gap blocks.
+			return Ok(())
+		}
+
```

Comment:
This is interesting. In what circumstances we could have both gap and forktree?

Reply:
this is a really bad hack and that is the motivation that I set as draft again. 
I would like to open a discussion about it. I'll ping you when I have more info